### PR TITLE
fix: fixed reduction parameter error

### PR DESCRIPTION
**Pull request by zac-williamson**
redc_param previously was only large enough to cover barrett reductions whose input was < 2^{modulus_bits * 2 + 2}

this was insufficient for elliptic curve arithmetic in bignum

redc_param is now large enough to cover barrett reduction inputs of at least 16 * modulus^2

additonally, library upgraded to not trigger compiler warnings for nargo 0.35.0

# Description

## Problem\*

Resolves 

## Additional Context

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: src/fields/U768.nr
Comment (TomAFrench) [inline src/fields/U768.nr:1]: why are the other field parameters changing here?

Comment (TomAFrench) [reply src/fields/U768.nr:1]: ah, this is just broken.

File: src/runtime_bignum_test.nr [hunk 1]
```diff
@@ -3,6 +3,194 @@ use crate::runtime_bignum::BigNumInstance;
...
+            [[X1, X1, X1]],
+            [[false, false, false]],
+            [],
+            []
+        )};
+        // if (CurveParams::a().get_limb(0) != 0) {
+        //     let ZZ = Z1.mul(Z1);
+        //     let AZZZZ = ZZ.mul(ZZ).mul(CurveParams::a());
+        //     XX_mul_3 = XX_mul_3.add(AZZZZ);
+        // }
```
Comment (vezenovm) [inline src/runtime_bignum_test.nr:97]:
```suggestion
```
leftover from debugger?

File: src/runtime_bignum_test.nr [hunk 2]
```diff
@@ -3,6 +3,194 @@ use crate::runtime_bignum::BigNumInstance;
...
+
+#[test]
+fn test_quadratic_expression() {
+    let bn = $typ ::get_instance();
+    for i in 0..32 {
+    let X1: BigNum<$N, $typ> = unsafe{ bn.__derive_from_seed([i as u8,2,3,4]) };
+    let Y1: BigNum<$N, $typ> = unsafe{ bn.__derive_from_seed([i as u8,i as u8,6,7]) };
+    let Z1: BigNum<$N, $typ> = BigNum::one();
+
+        let (_, YY_mul_2): (BigNum<$N, $typ>, BigNum<$N, $typ> ) = unsafe{BigNum::__compute_quadratic_expression([[Y1]], [[false]], [[Y1, Y1]], [[false, false]], [], [])};
```
Comment (vezenovm) [inline src/runtime_bignum_test.nr:84]: file could use a `nargo fmt`. Although, I know the formatter struggles with macros.

File: src/runtime_bignum_test.nr [hunk 3]
```diff
@@ -67,10 +238,27 @@ type Fq = BigNum<3, BNParams>;
...
 
+fn test_derive<let N: u32, Params>(BNInstance: BigNumInstance<N, Params>) where Params: BigNumParamsTrait<N> {
+    let a: BigNum<N, Params> = unsafe {
+        BNInstance.__derive_from_seed("hello".as_bytes())
+    };
+    let b: BigNum<N, Params> = unsafe {
+        BNInstance.__derive_from_seed("hello".as_bytes())
+    };
+    assert(BNInstance.eq(a, b));
+    //  println(f"derived value = {a}");
```
Comment (vezenovm) [inline src/runtime_bignum_test.nr:249]:
```suggestion
```

File: src/runtime_bignum_test.nr [hunk 4]
```diff
@@ -3,6 +3,194 @@ use crate::runtime_bignum::BigNumInstance;
...
+        let (_, Z3): (BigNum<$N, $typ>, BigNum<$N, $typ> ) = unsafe{ BigNum::__compute_quadratic_expression([[Y1]], [[false]], [[Z1, Z1]], [[false, false]], [], []) };
+
+        bn.validate_in_field(X3);
+        bn.validate_in_field(Y3);
+        bn.validate_in_field(Z3);
+    }
+}
+
+#[test]
+fn assert_is_not_equa() {
```
Comment (vezenovm) [inline src/runtime_bignum_test.nr:126]:
```suggestion
fn assert_is_not_equal() {
```

File: src/runtime_bignum_test.nr [hunk 5]
```diff
@@ -123,40 +319,62 @@ fn test_add<let N: u32, Params>(bn: BigNumInstance<N, Params>) where Params: Big
...
+        };
+        let result = unsafe {
+            bn.__mul(u, v)
+        };
         let expected: BigNum<N, Params> = BigNum::one();
         assert(result.limbs == expected.limbs);
     }
 }
 
 fn assert_is_not_equal<let N: u32, Params>(bn: BigNumInstance<N, Params>) where Params: BigNumParamsTrait<N> {
```
Comment (vezenovm) [inline src/runtime_bignum_test.nr:349]: Is this func used anywhere now that we have `assert_is_not_equal` in the macro test with 
```
#[make_test(quote{3},quote{BNParams})]
pub struct BNTests{}
```
Seems we can delete some of these functions that take in a `BigNumInstance` as inputs.

**Review by vezenovm — APPROVED** [review]
Looks good, mostly nits

**Pull request merged**